### PR TITLE
Improve performance on summing adjustments and payment

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

`:amount` is a database field in these cases, as opposed to a method that returns a computed result. Calling `.sum(:amount)` is much more efficient as it computes the sum at database level, as opposed to `.map(&:amount).sum`, which would fetch and instanciate all the objects first, and then sum the amounts after.

#### What should we test?

Green build.

#### Release notes

Improved performance on summing payment and adjustment amounts.

Changelog Category: Technical changes

### CODE REVIEW

File: app/helpers/checkout_helper.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def checkout_adjustments_for(order, opts = {})
     adjustments.reject! { |a| a.originator_type == 'EnterpriseFee' && a.source_type != 'Spree::LineItem' }
     unless exclude.include? :admin_and_handling
       adjustments << Spree::Adjustment.new(
-        label: I18n.t(:orders_form_admin), amount: enterprise_fee_adjustments.map(&:amount).sum
+        label: I18n.t(:orders_form_admin), amount: enterprise_fee_adjustments.sum(:amount)
```
Comment (mkllnk) [inline app/helpers/checkout_helper.rb:22]: I think that your assumption of having an AR relation isn't correct here. `adjustments` are an array and `enterprise_fee_adjustments` are an array, too. You can simplify this though:
```suggestion
        label: I18n.t(:orders_form_admin), amount: enterprise_fee_adjustments.sum(&:amount)
```

**Review by mkllnk — CHANGES_REQUESTED** [review]
Nice! This should really speed things up. I found one case though where we have an array and can't do this.

**Comment by mkllnk** [comment]
> What should we test?
> 
> Green build.

Is that a new strategy? It's probably good to take some load off our testers but it looks like our code coverage isn't quite good enough for this. Did I miss a discussion here?

**Comment by Matt-Yorkley** [comment]
Nice catch @mkllnk !

> Is that a new strategy?

Not in general, but I've been doing it for very small changes that seem to be covered by specs. I missed this one :see_no_evil:

**Review by mkllnk — APPROVED** [review]

**Review by sauloperez — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

Quite a noticeable improvement :tada: 

Clicking "Update and Recalculate Fees" on this bulky order:

Before:
image

After:
image

The values match, ready to go.

**Comment by sauloperez** [comment]
wow! :tada:

**Comment by Matt-Yorkley** [comment]
`9.37 seconds -> 3.12 seconds`

Hot stuff, that's a ~67% reduction! I didn't realise the impact would be that big. There should be multiple places in the app where we get this kind of boost.

:tada: :smile:

**Pull request merged**